api/helpers.js

```diff
@@ -333,19 +333,7 @@ const processPartnershipRecord = r => {
       occupation: block(r.partner1.occupation),
       retired: block(r.partner1.retired),
       address: block(r.partner1.address),
-      aliases: [
-        {
-          prefix: block(r.partner1.aliases[0].prefix),
-          forenames: block(r.partner1.aliases[0].forenames),
-          surname: block(r.partner1.aliases[0].surname),
-          suffix: block(r.partner1.aliases[0].suffix)
-        }, {
-          prefix: block(r.partner1.aliases[1].prefix),
-          forenames: block(r.partner1.aliases[1].forenames),
-          surname: block(r.partner1.aliases[1].surname),
-          suffix: block(r.partner1.aliases[1].suffix)
-        }
-      ],
```

Comment:
@RobinKnipe: Do you remember why it was done this way?

I can't understand how this could have worked for most records. (Maybe we only ever tested it with records that had aliases on them? Or maybe the wrangler always produces an array of 2 even if the fields are null?)

Reply:
Hmmm, not sure, it definitely doesn't look right though, as you say I would have expected errors from this! I kind of think that the "blocking" should be done more programmatically and much DRYer...
```js
const blockObj = obj => Object.fromEntries(Object.entries(obj).map(([key, value]) =>
    Array.isArray(value) ? [key, value.map(blockObj)] :
    value === Object(value) ? [key, blockObj(value)] :
    [key, 'BLOCKED']
));
const block = ({ id, blocked, ...record }) => ({ id, blocked, ...(blocked ? blockObj(record) : record) });
// then call it with:
const possiblyBlockedRecord = block(someRecord);
```
The *major advantages* of this approach are:
 - a single function (and recursive subfunction) for ALL datatypes
 - massive copy/paste code kull
 - any spec changes are automatically covered, i.e.
    - deleted fields, _no problem_
    - added fields, _automatically supported_
 - only 2 functions need testing, not 2 scenarios for each datatype!
 - block/unblocking fields is as simple as removing/adding them to the list of params in the `block` function
 - naturally defensively programmed, and can increase blocking potential with the block-by-default test:
```js
  blocked !== false ? blockObj(record) : record
```

Reply:
I think this assumes that we are censoring the API response and sending it on.

What this function does is take the API response and turns it into something that the view can understand. So for example statuses are turned into flags, and blocked records get `nulls` turned into the string, `'BLOCKED'`.

I'm not sure I would have written it that way but there is something to be said for not having to change the view just because the API changed.

> any spec changes are automatically covered

This would simply mean that things fail further down, in the view.

Reply:
Something like this would be better placed in the API. But I don't even think that is a good idea until we have TypeScript. The reason is that we want to have a guarantee that the API will spit out the correct schema. i.e. We want the more verbose form until we can enforce the schema via a type system.